src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,103 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int LENGTH_ARRAY = 10;
+    private static final double LENGTH_GROW = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[LENGTH_ARRAY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size() >= array.length) {
```

Comment:
```suggestion
        if (size >= array.length) {
```
You have access to the size variable without having to call the method